Problem-2

```diff
@@ -0,0 +1,79 @@
+
+289. Game of Life
+
+According to Wikipedia's article: "The Game of Life, also known simply as Life, is a cellular automaton devised by the British mathematician John Horton Conway in 1970."
+
+The board is made up of an m x n grid of cells, where each cell has an initial state: live (represented by a 1) or dead (represented by a 0). Each cell interacts with its eight neighbors (horizontal, vertical, diagonal) using the following four rules (taken from the above Wikipedia article):
+
+Any live cell with fewer than two live neighbors dies as if caused by under-population.
+Any live cell with two or three live neighbors lives on to the next generation.
+Any live cell with more than three live neighbors dies, as if by over-population.
+Any dead cell with exactly three live neighbors becomes a live cell, as if by reproduction.
+The next state is created by applying the above rules simultaneously to every cell in the current state, where births and deaths occur simultaneously. Given the current state of the m x n grid board, return the next state.
+
+ 
+// Time Complexity :O(nxm)
+// Space Complexity :O(1)
+// Did this code successfully run on Leetcode : No, not running on leetcode, can you please help fix the problem, i cant seem to find what is wrong.
+
+board =
+[[0,1,0],[0,0,1],[1,1,1],[0,0,0]]
+Output
+[[0,0,0],[0,0,0],[0,0,0],[0,0,0]]
+Expected
+[[0,0,0],[1,0,1],[0,1,1],[0,1,0]]
+
+
+
+
+class Solution {
+public:
+  
+    int dirarr[8][2] = {{0,1}, {0,-1}, {1,0}, {-1,0}, {-1,-1}, {-1,1}, {1,-1}, {1,1} };
+
+    int count(vector<vector<int>>& board, int i, int j, int n, int m) {
+        int countneigh =0;
+
+        for(int k=0; k< 8; k++){
+            int row = i+dirarr[k][0];
+            int col = j+dirarr[k][1];
+            if ((row >=0) && (col >= 0) && (row < n-1) && (col < m-1)) {
+                if ((board[row][col] == 1) || (board[row][col] == 3)) {
```

Comment:
condition should be ` if ((board[row][col] == 1) || (board[row][col] == 2))` because 2 means previously live cell and 3 means previously dead cell.